eg/Classes/PluginBase.py

```diff
@@ -148,7 +148,15 @@ def AddActionsFromList(self, theList, defaultAction=None):
         self.info.actionGroup.AddActionsFromList(theList, defaultAction)
 
     def AddEvents(self, *eventList):
-        self.info.eventList = eventList
+        if self.info.eventList is None:
+            self.info.eventList = ()
+
+        for event in eventList:
+            if isinstance(event, basestring):
+                event = (event, '')
+            elif isinstance(event, list):
+                event = tuple(event)
+            self.info.eventList += (event,)
```

Comment:
Instead of redefining `event`, directly add to `eventList`?
```python
        for event in eventList:
            if isinstance(event, basestring):
                self.info.eventList += ((event, ''),)
            elif isinstance(event, list):
                self.info.eventList += (event, )
```